test/set.js

```diff
@@ -22,6 +22,22 @@ describe('set', function() {
     assert.deepEqual(b, {w: 1, x: {a: 5, b: 2, c: 3}, z: 4});
   });
 
+  it('handles empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({}, a);
+    assert.deepEqual(b, a);
+  });
+
+  it('handles nested empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({w: 0, x: {}, z: 3}, a);
+    assert.deepEqual(b, {w: 0, x: {a: 1, b: 2, c: 3}, z: 3});
+  });
+
```

Comment:
This becomes tricky here.  It's not so obvious to me that this is the desired result.  I think there's a good case to be made for

    {w: 0, x: {}, z: 3}

Or put another way, if the user want to use `set` to transform `{w: 1, x: {a: 1, b: 2, c: 3}, z: 4}` to ` {w: 0, x: {}, z: 3}`, what should she pass as the first parameter?